internal/setup/setup.go

```diff
@@ -678,3 +755,35 @@ func Select(release *Release, slices []SliceKey) (*Selection, error) {
 
 	return selection, nil
 }
+
+// essentialSlices returns a list of slices that must be installed. These
+// include the slice which contains the "generate: chisel-state" path.
+func essentialSlices(release *Release) []*Slice {
+	if release == nil {
+		return []*Slice{}
+	}
+
+	essential := []*Slice{}
+	// Add the slice with "generate: chisel-state" property.
+	stateSlice, _, err := release.FindChiselState()
+	if err == nil {
+		essential = append(essential, stateSlice)
+	}
+	return essential
+}
+
+// FindChiselStateSlice returns the slice which contains the path with
+// "generate: chisel-state" property and the respective path. If there is no
+// such slice, return nil.
+func (r *Release) FindChiselState() (*Slice, string, error) {
+	for _, pkg := range r.Packages {
+		for _, slice := range pkg.Slices {
+			for path, info := range slice.Contents {
+				if info.Kind == GeneratePath && info.Generate == GenerateState {
+					return slice, path, nil
```

Comment:
Do we have something to validate that there is only one?

Reply:
N/A due to recent discussions.